schema

```diff
@@ -206,16 +199,22 @@
         }
       },
       "not": {
+        "$comment": "This is what the specification requires, but it seems bad. https://github.com/json-api/json-api/issues/1553",
         "anyOf": [
-          {"required": ["relationships"]},
-          {"required": ["links"]},
-          {"required": ["id"]},
-          {"required": ["type"]}
```

Comment:
would this then allow "id" and "type" as `attributes` member, which violates one of the [restrictions for fields](https://jsonapi.org/format/#document-resource-object-fields)?

> a resource can not have an attribute and relationship with the same name, nor can it have an attribute or relationship named **type** or **id**.